File: api/src/main/java/hirez/api/object/statuspage/Summary.java
package hirez.api.object.statuspage;

import lombok.Data;

import java.util.List;

@Data
public class Summary {
    private final StatusData status;
    private final List<Component> components;
    private final List<Incident> incidents;
    private final List<ScheduleMaintenance> scheduledMaintenances;
    private final Page page;
}

File: api/src/main/kotlin/hirez/statuspage/json/Summary.kt
package hirez.statuspage.json

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Summary(
			override val status: StatusData,
			override val components: Array<Component>,
			override val incidents: Array<Incident>,
			override val scheduledMaintenances: Array<ScheduleMaintenance>,
			override val page: StatusPage
) : IStatusPage, IStatus,
			IComponents, IIncidents,
			IScheduledMaintenances